esp/src/ganglia/GangliaWidget.js

```diff
@@ -0,0 +1,422 @@
+            this.clusterTargetSelect.required = false;
+            WsRrd.GangliaClusterList({
+                request:{
+                    Cluster:"",
+                    Server:""
+                },
+            }).then(function (response) {
+                if (lang.exists("GetAvailableClustersResponse.Clusters.Item", response)) {
+                    var output = arrayUtil.map(response.GetAvailableClustersResponse.Clusters.Item, function(item, idx) {
+                        return {
+                            name: item,
+                            id: item
+                        };
+                    });
+                    var myStore = new Memory({
+                        data: output
+                    });
+                    context.clusterTargetSelect.set("store", myStore);
+                }
+            });
+        },
+
+        _buildServers: function (newValue) {
+            var context = this;
+            console.log(this.serverTargetSelect);
+            this.serverTargetSelect.required = false;
+            WsRrd.GangliaServerList({
+                request:{
+                    Cluster: this.cluster
+                },
+            }).then(function (response) {
+                if (lang.exists("GetAvailableServersForMetricsResponse.Servers.Item", response)) {
+                    var output = arrayUtil.map(response.GetAvailableServersForMetricsResponse.Servers.Item, function(item, idx) {
+                        return {
+                            name: item,
+                            id: item
+                        };
+                    });
+                    var myStore = new Memory({
+                        data: output
+                    });
+                    context.serverTargetSelect.set("store", myStore);
+                }
+            });
+        },
+
+        _buildMetrics: function (newValue) {
+            var context = this;
+            this.metricsTargetSelect.required = true;
+            WsRrd.GangliaMetricList({
+                request:{
+                    Cluster:this.cluster,
+                    Server:this.server
+                },
+            }).then(function (response) {
+                if (lang.exists("GetAvailableMetricsResponse.Metrics.Item", response)) {
+                    var output = arrayUtil.map(response.GetAvailableMetricsResponse.Metrics.Item.sort(), function(item, idx) {
+                        return {
+                            name: item,
+                            id: item
+                        };
+                    });
+
+                    var myStore = new Memory({
+                        data: output
+                    });
+                    context.metricsTargetSelect.set("store", myStore);
+                }
+            });
+        },
+
+        _buildTabs: function () {
+            var context = this;
+            var tabs = registry.byId(this.id + "TabContainer");
+            xhr("esp/files/ganglia/ganglia.json", {
```

Comment:
I wonder if that should be "/esp/files/ganglia/ganglia.json" - additional "/" at the start?